content/issue-1/minilsm.md

```diff
@@ -0,0 +1,125 @@
+2. Write the key-value pair to memtable. After (1) and (2) completes, we can notify the user that the write operation
+   is completed.
+3. When a memtable is full, we will flush it to the disk as an SST file in the background.
+4. We will compact some files in some level into lower levels to maintain a good shape for the LSM tree, so that read
+   amplification is low.
+
+## Read Flow
+
+![](https://skyzh.github.io/mini-lsm/figures/lsm-tutorial/00-lsm-read-flow.svg)
+
+When we want to read a key,
+
+1. We will first probe all the memtables from latest to oldest.
+2. If the key is not found, we will then search the entire LSM tree containing SSTs to find the data.
+
+## Tutorial Overview
+
+![](https://skyzh.github.io/mini-lsm/figures/lsm-tutorial/00-lsm-tutorial-overview.svg)
+
+As of the end of 2022, we have finished the first 4 parts of the tutorial.
+
+### Day 1 - Block Encoding
+
+Block is the minimum read unit in LSM. It is of 4KB size in general, similar database pages. In each block, we will
+store a sequence of sorted key value pairs. In day 1, we will explore how to use the `bytes` library to do manual data
+encoding. Also, you will learn how to write RocksDB-style iterator in Rust.
+
+### Day 2 - Table Encoding
+
+SST (sorted string table) is composed of data blocks and index blocks stored on the disk. Usually, data blocks are
+lazily loaded -- they will not be loaded into the memory until a user requests it. Index blocks can also be loaded
+on-demand, but in this tutorial, we make simple assumptions that all SST index blocks (meta blocks) can fit in memory.
+Generally, an SST file is of 256MB size. In day 2, we will implement SST encoding and learn how to compose RocksDB-style
+iterators together in Rust.
+
+### Day 3 - Mem Table and Merge Iterators
+
+In this tutorial, we use [crossbeam-skiplist](https://docs.rs/crossbeam-skiplist) as the implementation of memtable. You
+will learn how to use self-referential struct in Rust to erase lifetime from a Rust-style iterator object, and therefore
+being easily composed with other RocksDB-style iterators in the system.
+
+Also, we will see how to implement high-performance merge iterators with correct error handling with Rust's `BinaryHeap`.
+We intentionally avoid using dynamic dispatch (i.e. `Box<dyn Iterator>`) to achieve higher performance. You will need to
+implement two different merge iterators: a two-merge iterator to merge different types of iterators, and a multi-way
+merge iterator to merge multiple iterators of the same type. You will have a better understanding of lifetime and trait
+system in Rust.
+
+### Day 4 - Storage Engine and Block Cache
+
+In this part, you will need to compose everything you have implemented in previous parts to build a fully functional
+storage engine. You will explore how surprisingly easy for us to maintain object references and ownerships in a
+multi-thread program with the help of borrow checker.
+
+## Upcoming Chapters
+
```

Comment:
Currently the tutorial both is and appears unfinished.  It would seem to be clearer for the tutorial to be a finished piece of work with no empty headings.  Can you tie it up for now so it has a more finished look to this point?